llzk_backend/src/template.rs

```diff
@@ -485,13 +531,18 @@ where
                 initializations.gen_llzk_in_template(codegen, template)
             }
             Statement::Declaration { meta, xtype, name, dimensions, .. } => {
-                // TODO: we've already handled declarations to create struct fields and function
-                // parameters. Is there any reason to visit them again? If not, then
-                // we don't need the InitializationBlock above either.
-                println!("TODO: anything else to do with declaration? {name} of type {xtype:?}");
-                Ok(())
+                template.and_then_same(codegen, |fc, _| {
+                    fc.block_ctx.declare_name_if_not_present(name, || {
+                        codegen.new_nondet_value_of_dimensions(meta, dimensions)
+                    })
+                })
```

Comment:
This will add declarations for any kind, but signals and such are handled in a different data structure, no? This should either exclude those or do only the `Var` case. Subcomponents will probably follow a different strategy and have different semantics